File: 程式設計：資料結構答案檔/Python/102.py
m = int(input())
n = int(input())

matrix = []
for i in range(m):
    row = []
    for j in range(n):
        row.append(int(input()))
    matrix.append(row)

print("Original:")
for row in matrix:
    print("[" + " ".join([str(value) for value in row]) + "]")

transpose_matrix = []
for i in range(n):
    row = []
    for j in range(m):
        row.append(matrix[j][i])
    transpose_matrix.append(row)

print("Transpose:")
for row in transpose_matrix:
    print("[" + " ".join([str(value) for value in row]) + "]")

File: 程式設計：資料結構答案檔/Go/102.go
package main

import (
	"fmt"
)

func main() {
	var m, n int
	fmt.Scan(&m)
	fmt.Scan(&n)

	matrix := make([][]int, m)
	for i := range matrix {
		matrix[i] = make([]int, n)
		for j := range matrix[i] {
			fmt.Scan(&matrix[i][j])
		}
	}

	fmt.Println("Original:")
	for _, row := range matrix {
		fmt.Print("[")
		for j, value := range row {
			if j > 0 {
				fmt.Print(" ")
			}
			fmt.Print(value)
		}
		fmt.Println("]")
	}

	transposeMatrix := make([][]int, n)
	for i := range transposeMatrix {
		transposeMatrix[i] = make([]int, m)
		for j := range transposeMatrix[i] {
			transposeMatrix[i][j] = matrix[j][i]
		}
	}

	fmt.Println("Transpose:")
	for _, row := range transposeMatrix {
		fmt.Print("[")
		for j, value := range row {
			if j > 0 {
				fmt.Print(" ")
			}
			fmt.Print(value)
		}
		fmt.Println("]")
	}
}
